utils/auto_dockerfile_update/update_dockerfiles.py

```diff
@@ -27,33 +30,205 @@ def is_docker_file_outdated(dockerfile: Dict, latest_tag: str, last_updated: str
+                f.write(file_content)
+            return True
+        return False
+
+
+def change_python_version(file_path: str, str_version: str) -> Tuple[bool, str]:
+    """Replace the current python version in the Pipfile or the pyproject.toml.
+
+    Args:
+        file_path (str): The file path to the Pipfile or the pyproject.toml file.
+        version (str): The version.
+        full_str_python_version (str): The older version.
+    """
+    (
+        current_version,
+        full_str_python_version,
+        success_extraction,
+    ) = extract_current_python_version(file_path)
+    if success_extraction:
+        result = replace_python_version(file_path, str_version, full_str_python_version)
+        return result, current_version
+    else:
+        print(f"[ERROR] can't extract python version form: {file_path}")
+    return False, ""
+
+
+def run_lock(base_path_docker: str, pipfile_or_pyproject_path: str) -> bool:
+    """Runs poetry lock --no-update or pipfile lock --keep-outdated.
+
+    Args:
+        base_path_docker (str): The DockerFile path.
+        pipfile_or_pyproject_path (str): The file path to the Pipfile
+        or the pyproject.toml file.
+    """
+    base_path = base_path_docker.replace("/Dockerfile", "")
+    current_directory = os.getcwd()
+    os.chdir(f"{current_directory}/" + base_path)
+    try:
+        if "Pipfile" in pipfile_or_pyproject_path:
+            # waits for the process to end.
+            completed_process = subprocess.run(
+                ["pipenv", "lock", "--keep-outdated"],
+                check=True,
+                stdout=subprocess.DEVNULL,
+                stderr=subprocess.PIPE,
+            )
+            if completed_process.returncode != 0:
+                return False
+        elif "pyproject.toml" in pipfile_or_pyproject_path:
+            completed_process = subprocess.run(
+                ["poetry", "lock", "--no-update"],
+                stderr=subprocess.PIPE,
+                check=True,
+                stdout=subprocess.DEVNULL,
+            )
+            if completed_process.returncode != 0:
+                os.chdir(current_directory)
+                return False
+    except subprocess.CalledProcessError as e:
+        print(f"[ERROR] Lock failed with error: {str(e.stderr)} for {base_path_docker}")
+        return False
+    except Exception as e:
+        print(f"[ERROR] {e}: for {base_path_docker} for {base_path_docker}")
+        os.chdir(current_directory)
+        return False
+    finally:
+        os.chdir(current_directory)
+    return True
+
+
+def update_python_version_pyproject_or_pipfile(
+    dockerfile: Dict, latest_tag: str
+) -> None:
+    """
+    Updating dockerfile content with the latest tag
+    Args:
+        dockerfile (Dict): Dockerfile dict.
+        latest_tag (str): latest tag string.
+    Returns:
+        None
+    """
+    update_result = False
+    image_name = dockerfile["image_name"]
+    if "python" in image_name:
```

Comment:
if python not in image_name:
return
rest of function not-nested.
Generally good practice to reduce nesting